c_src/ep_encoder.c

```diff
@@ -675,8 +675,6 @@ get_oneof_field(ErlNifEnv *env, ERL_NIF_TERM term, ep_node_t *node, ERL_NIF_TERM
         return NULL;
     }
 
-    *out = term;
-
     unsigned size = 100;
     char *buf = _alloc(size);
```

Comment:
I am not sure if following is the same. If it is, consider using the following and you will not need _free(buf).
```suggestion
    char buf[100];
```